src/core/prometheus.cc

```diff
@@ -217,9 +217,9 @@ static std::ostream& operator<<(std::ostream& os, const seastar::metrics::impl::
     switch (v.type()) {
     case seastar::metrics::impl::data_type::GAUGE:
     case seastar::metrics::impl::data_type::REAL_COUNTER:
-        return os << v.d();
+        return os << std::to_string(v.d());
     case seastar::metrics::impl::data_type::COUNTER:
-        return os << v.i();
+        return os << std::to_string(v.i());
```

Comment:
And for symmetry:
```suggestion
        fmt::print(os, "{}", v.i());
        return os;
```
which may have the advantage of not allocating for large numbers.